cmd/storagenode/deprecated.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// Deprecated contains deprecated config structs
+type Deprecated struct {
+	Kademlia struct {
+		ExternalAddress string `default:"" hidden:"true"`
+		Operator        struct {
+			Email  string `default:"" hidden:"true"`
+			Wallet string `default:"" hidden:"true"`
+		}
+	}
+}
+
+// maps deprecated config values to new values if applicable
+func mapDeprecatedConfigs(log *zap.Logger) {
+	type migration struct {
+		newValue        interface{}
+		newConfigString string
+		oldValue        string
+		oldConfigString string
+	}
+	migrations := []migration{
+		{
+			newValue:        &runCfg.Contact.ExternalAddress,
+			newConfigString: "contact.external-address",
+			oldValue:        runCfg.Deprecated.Kademlia.ExternalAddress,
+			oldConfigString: "kademlia.external-address",
+		},
+		{
+			newValue:        &runCfg.Operator.Wallet,
+			newConfigString: "operator.wallet",
+			oldValue:        runCfg.Deprecated.Kademlia.Operator.Wallet,
+			oldConfigString: "kademlia.operator.wallet",
+		},
+		{
+			newValue:        &runCfg.Operator.Email,
+			newConfigString: "operator.email",
+			oldValue:        runCfg.Deprecated.Kademlia.Operator.Email,
+			oldConfigString: "kademlia.operator.email",
+		},
+	}
+
+	for _, migration := range migrations {
+		if migration.oldValue != "" {
+			typ := reflect.TypeOf(migration.newValue).Elem()
+			override := parseOverride(typ, migration.oldValue)
+
+			reflect.ValueOf(migration.newValue).Elem().Set(reflect.ValueOf(override))
+
+			log.Sugar().Debugf("Found deprecated flag. Migrating value %v from %s to %s", reflect.ValueOf(migration.newValue).Elem(), migration.oldConfigString, migration.newConfigString)
```

Comment:
change debug to info.
add link to a "more info" document describing any config renames

Reply:
Add comment in doc that the config will pick up the old value?